python/cugraph-service/tests/test_e2e.py

```diff
@@ -515,6 +515,29 @@ def test_uniform_neighbor_sampling(client_with_edgelist_csv_loaded):
     )
 
 
+def test_inside_asyncio_event_loop(client_with_edgelist_csv_loaded):
```

Comment:
Can you parameterize this test so `result_device` is tested with both a device ID and `None`?  That will ensure the same code paths work from an `async` caller both when the client `async` functions are used and not used. 
Also I've typically put tests for device data transfer in `test_mg_e2e.py` to ensure transfers happen between different GPUs as opposed to from GPU 0 to GPU 0 (although either should still exercise the same code paths, I think).